packages/hardhat-plugin/src/plan/utils.ts

```diff
@@ -1,30 +1,41 @@
-import {
+import type {
   VertexGraph,
   VertexDescriptor,
-  IgnitionError,
+  DeploymentGraphVertex,
 } from "@ignored/ignition-core";
 
-export function parseType(v: VertexDescriptor): string {
-  let type: string;
-  switch (v.type) {
+type DisplayType =
+  | "deploy-contract"
+  | "deploy-library"
+  | "call"
+  | "transfer"
+  | "event";
+
+export function parseType(v: VertexDescriptor): DisplayType {
+  const depVertex = v as DeploymentGraphVertex;
+  switch (depVertex.type) {
     case "HardhatContract":
     case "ArtifactContract":
-      type = "deploy";
-      break;
+    case "DeployedContract":
```

Comment:
I'm not sure it's appropriate to handle `DeployedContract`s the same as the other two since it doesn't create a transaction or interact with the chain at all

Reply:
I kind of think they shouldn't be on the `plan` diagram at all

Reply:
Agreed. We shouldn't be seeing the deployed contract in the execution graph as nothing is executed.

I think my ideal would be to include the deployed contract vertex in the deployment graph and validate it (that the address makes sense, maybe at some point in the future that the on chain address exists and the abi matches or some other useful check?), then eliminate it in the execution graph.

I can think of two approaches to the elimination. In transformation we simplify the deployment graph after validation to create the execution graph. The execution of deployed contract just passes the input abi/address to the result context. To replicate this we could:

1. Modify transform to allow prepopulating the result context, so pull the deployed contract logic currently in execution forward to the graph transform step
2. Change future result resolution so that if a deployed contract future is passed in, the abi/address is resolved from the data on the future rather than from the result context.

I am tempted by 2 currently, any thoughts?

Reply:
I am going to separate this out into its own issue: https://github.com/NomicFoundation/ignition/issues/170

I think the PR does fix the lib display issue, it is just that it has led us to thinking about what is displayed across all vertex types.